### PR TITLE
Multiple categories

### PR DESCRIPTION
**Pull request by btseytlin**
Multiple categories are detected in the following way:
* Column consists of comma-delimetered tags like 'category1, category2, category3' (spaces get stripped)
* The median number of tags if split by commas is greater than 1
* The number of unique tags is between 5 and 30

I tried to make it work with arrays or lists, but it ended up breaking too many things 

Other changes:
* Tokenization of text now applies lowercasing, so that the resulting vocab is smaller. This way the one-hot/cae encoders have smaller input dimensionality as well, which speeds stuff up
* Fixed the `showcapture` plugin, now pytest should actually hide warnings unless you need them
* Added a warning in case deduplication removes way too many rows

### CODE REVIEW

File: tests/integration_tests/data_sources/test_clickhouse_ds.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import pytest
 import requests
 from mindsdb_native import Predictor
-from mindsdb_native.libs.data_sources.clickhouse_ds import ClickhouseDS
 from mindsdb_native import F
 
 
 @pytest.mark.integration
 def test_clickhouse_ds():
+    from mindsdb_native.libs.data_sources.clickhouse_ds import ClickhouseDS
```
Comment (George3d6) [inline tests/integration_tests/data_sources/test_clickhouse_ds.py]: Is this related to this PR ?

File: tests/integration_tests/data_sources/test_maria_ds.py [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 from mindsdb_native import Predictor
 from mindsdb_native.libs.constants.mindsdb import DATA_TYPES, DATA_SUBTYPES
-from mindsdb_native import MariaDS
 from mindsdb_native import F
 
 
 @pytest.mark.integration
 def test_maria_ds():
+    import mysql.connector
+    from mindsdb_native import MariaDS
```
Comment (George3d6) [inline tests/integration_tests/data_sources/test_maria_ds.py]: Is this related to this PR ?

File: tests/integration_tests/data_sources/test_mongodb_ds.py [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-
 from mindsdb_native import Predictor
-from mindsdb_native.libs.data_sources.mongodb_ds import MongoDS
 from mindsdb_native import F
 
 
 @pytest.mark.integration
 def test_mongodb_ds():
+    from pymongo import MongoClient
+    from mindsdb_native.libs.data_sources.mongodb_ds import MongoDS
```
Comment (George3d6) [inline tests/integration_tests/data_sources/test_mongodb_ds.py]: Is this related to this PR ?

File: tests/integration_tests/data_sources/test_ms_sql_ds.py [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 import logging
 from mindsdb_native import Predictor
-from mindsdb_native.libs.data_sources.ms_sql_ds import MSSQLDS
 from mindsdb_native import F
 
 
 @pytest.mark.integration
 def test_mssql_ds():
+    import pytds
+    from mindsdb_native.libs.data_sources.ms_sql_ds import MSSQLDS
```
Comment (George3d6) [inline tests/integration_tests/data_sources/test_ms_sql_ds.py]: Is this related to this PR ?

File: tests/integration_tests/data_sources/test_mysql_ds.py [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
-import mysql.connector
 import logging
 from mindsdb_native import Predictor
-from mindsdb_native.libs.data_sources.mysql_ds import MySqlDS
 from mindsdb_native import F
 
 @pytest.mark.integration
 def test_mysql_ds():
+    import mysql.connector
+    from mindsdb_native.libs.data_sources.mysql_ds import MySqlDS
```
Comment (George3d6) [inline tests/integration_tests/data_sources/test_mysql_ds.py]: Is this related to this PR ?

File: tests/integration_tests/data_sources/test_postgres_ds.py [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
-import pg8000
 from mindsdb_native import Predictor
-from mindsdb_native.libs.data_sources.postgres_ds import PostgresDS
 from mindsdb_native import F
 
 
 @pytest.mark.integration
 def test_postgres_ds():
+    import pg8000
+    from mindsdb_native.libs.data_sources.postgres_ds import PostgresDS
```
Comment (George3d6) [inline tests/integration_tests/data_sources/test_postgres_ds.py]: Is this related to this PR ?

File: mindsdb_native/libs/phases/data_cleaner/data_cleaner.py [hunk 7]
```diff
@@ -26,6 +28,23 @@ def _remove_duplicate_rows(self, df):
         if no_dropped > 0:
             self.log.warning(f'Dropped {no_dropped} duplicate rows.')
 
+    def _convert_iterables_to_tuples(self, df):
```
Comment (George3d6) [inline mindsdb_native/libs/phases/data_cleaner/data_cleaner.py]: Why is this required ?

Comment (George3d6) [reply mindsdb_native/libs/phases/data_cleaner/data_cleaner.py]: As in, I assume this converts arrays into tuple, but why is this needed ?

File: mindsdb_native/libs/phases/data_cleaner/data_cleaner.py [hunk 8]
```diff
@@ -1,3 +1,5 @@
+import six
+import numpy as np
```
Comment (George3d6) [inline mindsdb_native/libs/phases/data_cleaner/data_cleaner.py]: unused

**Review by George3d6 — CHANGES_REQUESTED** [review]
I know the PR is still WIP, but please try to separate everything that is not related to it (e.g. changes in where stuff is imported for the databases test) into a separate PR.

File: mindsdb_native/libs/phases/data_cleaner/data_cleaner.py [hunk 9]
```diff
@@ -41,7 +41,12 @@ def run(self):
 
         self._remove_missing_targets(df)
 
+        len_before_dedupe = len(df)
         if self.transaction.lmd.get('deduplicate_data'):
-            self._remove_duplicate_rows(df)
+           self._remove_duplicate_rows(df)
```
Comment (maximlopin) [inline mindsdb_native/libs/phases/data_cleaner/data_cleaner.py]: must be 4 spaces (in context of this `if` statement) before `self._remove_duplicate_rows(df)`
(might cause indentation error)

File: mindsdb_native/libs/phases/type_deductor/type_deductor.py [hunk 10]
```diff
@@ -90,22 +93,23 @@ def type_check_date(element):
...
                     if not get_number_subtype(ele):
                         all_nr = False
                         break
 
-                if all_nr:
-                    additional_info['separator'] = char
-                    type_guess = DATA_TYPES.SEQUENTIAL
-                    subtype_guess = DATA_SUBTYPES.ARRAY
+                if len(ele_arr) > 1 and all_nr:
+                        additional_info['separator'] = sep_char
```
Comment (maximlopin) [inline mindsdb_native/libs/phases/type_deductor/type_deductor.py]: weird indentation again

**Review by maximlopin — APPROVED** [review]

**Pull request merged**